# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBObjectStore.cpp
# Commit: 5be5a3bf0499
# Full Hash: 5be5a3bf0499f5501bf1ed6291ba1ac677eacbc6
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-11 16:05:23
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Reduce code duplication for different open cursor variants. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40956
# ==============================================================================

diff -r 6078d5a9c010 -r 5be5a3bf0499 dom/indexedDB/IDBObjectStore.cpp
--- a/dom/indexedDB/IDBObjectStore.cpp	Tue Sep 10 15:36:09 2019 +0000
+++ b/dom/indexedDB/IDBObjectStore.cpp	Tue Sep 10 15:36:15 2019 +0000
@@ -2270,22 +2270,14 @@
 
   IDBCursor::Direction direction = IDBCursor::ConvertDirection(aDirection);
 
-  OpenCursorParams params;
-  if (aKeysOnly) {
-    ObjectStoreOpenKeyCursorParams openParams;
-    openParams.commonParams().objectStoreId() = objectStoreId;
-    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.commonParams().direction() = direction;
-
-    params = std::move(openParams);
-  } else {
-    ObjectStoreOpenCursorParams openParams;
-    openParams.commonParams().objectStoreId() = objectStoreId;
-    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.commonParams().direction() = direction;
-
-    params = std::move(openParams);
-  }
+  const CommonOpenCursorParams commonParams = {
+      objectStoreId, std::move(optionalKeyRange), direction};
+
+  // TODO: It would be great if the IPDL generator created a constructor
+  // accepting a CommonOpenCursorParams by value or rvalue reference.
+  const auto params =
+      aKeysOnly ? OpenCursorParams{ObjectStoreOpenKeyCursorParams{commonParams}}
+                : OpenCursorParams{ObjectStoreOpenCursorParams{commonParams}};
 
   RefPtr<IDBRequest> request = GenerateRequest(aCx, this);
   MOZ_ASSERT(request);
